### ISSUE TITLE
App crash after reinstall

### ISSUE THREAD

**Issue by vafaminato** [android/play-billing-samples#303]
The app crashes after reinstall. First, user subscribes to a plan. Then I reinstall the app and open the app. App crashes before logging in. Stack trace log:

`java.nullpointerexception: attempt to invoke virtual method 'int java.lang.ordinal()' on a null object reference at {package}data.network.firebase.serverFunctionsImpl$4.onComplete(ServerFunctuinsImpl.java:328)`

**Comment by vafaminato** [comment]
Please someone tell us whether the classytaxyjava or classytaxykotlin works or not. This way at least we know we are doing it the wrong way. I already deleted everything (firebase project, android app from console, etc.) and I am starting it over hope to work again. Who is in charge of this? It seems just #calren is visiting and closing the issues.

**Comment by vafaminato** [comment]
I wrapped the error from firebase into an if-else statement and apparently the problem is solved. But I think this is not the permanent way to solve this issue.

**Comment by LordByron1788** [comment]
> Who is in charge of this? It seems just #calren is visiting and closing the issues.

It seems the only issues being closed on this GitHub repo are ones that don't actually require any work, or ones where no work is being done. No progress is being made on anything else. This was the case even before the current situation going on in the world. Billing is a business critical process, but that is not reflected in the quality of the samples, or how this Github repo is being managed. Both are woefully inadequate, and when complaints are made they are ignored and shut down. I am not suprised ClassyTaxiJava has issues, it was a rushed piece of work that was left unfinished. Also, it was used to close a completely unrelated github issue #260 . The community asked for a replacement for java Trivial Drive, and instead something different was delivered because the developer did not the understand requirements.

**Comment by vafaminato** [comment]
> Billing is a business critical process, but that is not reflected in the quality of the samples, or how this Github repo is being managed. Both are woefully inadequate, and when complaints are made they are ignored and shut down. I am not suprised ClassyTaxiJava has issues, it was a rushed piece of work that was left unfinished. Also, it was used to close a completely unrelated github issue #260 .

I think we all need this backend server anyway. This is my first time selling my app in google play. I really don't know why people are asking for server-less samples. It seems it simply is not secure. Please correct my answer if you think otherwise.

**Comment by LordByron1788** [comment]
> > Billing is a business critical process, but that is not reflected in the quality of the samples, or how this Github repo is being managed. Both are woefully inadequate, and when complaints are made they are ignored and shut down. I am not suprised ClassyTaxiJava has issues, it was a rushed piece of work that was left unfinished. Also, it was used to close a completely unrelated github issue #260 .
> 
> I think we all need this backend server anyway. This is my first time selling my app in google play. I really don't know why people are asking for server-less samples. It seems it simply is not secure. Please correct my answer if you think otherwise.

Billing is a business critical process. There should be a Java samples for both server and server-less implementation of the latest billing API. Currently, there are Kotlin versions for both of these. At the start of the year, there were no samples for Java of either of them using the new API. An explicit request was made for a "server-less" java sample, towards the end of the last year. This was a popular request based on "likes". Here are some related requests: #238 #260 #277 and the new one #278. There were no explicit requests for a server based example at the time. But of course it is also going to be needed.  So based on this demand, if you are updating samples, it would make sense to tackle the server less example first. If you are not going to do that, then Google should NOT have communicated on those Github issues above stating that they were working on a Java sample. It was total miscommunication, and it has wasted time planning for this effort. They should have created another issue separate, and stated their intention to deliver a server based product only along the lines of ClassyTaxi.  At least then we would have had early notification of this. Instead, we got weeks of miscommunication. Closing off all those issues for Trivial Drive after promising a sample, with a java ClassyTaxi version is just poor management. As for the point of everybody using server-less. It is not possible for everybody to go down this route.  As long as the TrivialDriveKotlin example remains and is supported, then there should be a java sample to compliment it. When you read issue #278 it is like, they completely ignored what was requested in previous issues.

**Comment by vafaminato** [comment]
> > > Billing is a business critical process, but that is not reflected in the quality of the samples, or how this Github repo is being managed. Both are woefully inadequate, and when complaints are made they are ignored and shut down. I am not suprised ClassyTaxiJava has issues, it was a rushed piece of work that was left unfinished. Also, it was used to close a completely unrelated github issue #260 .
> > 
> > 
> > I think we all need this backend server anyway. This is my first time selling my app in google play. I really don't know why people are asking for server-less samples. It seems it simply is not secure. Please correct my answer if you think otherwise.
> 
> Billing is a business critical process. There should be a Java samples for both server and server-less implementation of the latest billing API. Currently, there are Kotlin versions for both of these. At the start of the year, there were no samples for Java of either of them using the new API. An explicit request was made for a "server-less" java sample, towards the end of the last year. This was a popular request based on "likes". Here are some related requests: #238 #260 #277 and the new one #278. There were no explicit requests for a server based example at the time. But of course it is also going to be needed. So based on this demand, if you are updating samples, it would make sense to tackle the server less example first. If you are not going to do that, then Google should NOT have communicated on those Github issues above stating that they were working on a Java sample. It was total miscommunication, and it has wasted time planning for this effort. They should have created another issue separate, and stated their intention to deliver a server based product only along the lines of ClassyTaxi. At least then we would have had early notification of this. Instead, we got weeks of miscommunication. Closing off all those issues for Trivial Drive after promising a sample, with a java ClassyTaxi version is just poor management. As for the point of everybody using server-less. It is not possible for everybody to go down this route. As long as the TrivialDriveKotlin example remains and is supported, then there should be a java sample to compliment it. When you read issue #278 (which they reluctantly created) it is like, they pretend this issue has never been asked before.

Thank you for your comment. Yes, somehow it took me one month to setup this purchase flow with the backend server. still there are some uncertainty about the stability of it. Sometimes, I think I have to choose serverless as it is less complicated and straight forward. Or google leaves us no choice... They are totally ignorant. Even when you message them in play console they come back after a week and refer to some android dev. tutorials.

**Comment by LordByron1788** [comment]
Yes, sometimes it is difficult to get Play Console Support to understand even the most basic of issues. They seem to lack the right level of training or understanding of how things work.
